### PR TITLE
Flutter doctor, condition not to check android ios toolchain

### PR DESCRIPTION
**Pull request by franzsilva**
#12768

### CODE REVIEW

**Comment by xster** [comment]
Thanks for your contribution! 

Instead of a specific bool flag, how about a general syntax like

`flutter doctor --only=fuchsia,ios`

**Comment by Hixie** [comment]
I think maybe `--fuchsia` is the way to go. Turn off Android and iOS, turn on Fuchsia. It should probably be a hidden command too since most people won't use it (in fact it's only really intended for scripts). I don't think there's much value in an --only flag since nobody's going to be saying `flutter doctor --only=android` or `flutter doctor --only=ios,fuchsia` or whatever.

**Comment by franzsilva** [comment]
Cool!, ill change that, is there any other thing it need to hide, like the IDE'S?, or just the sdk's?

**Comment by xster** [comment]
If the user isn't checking for the android toolchain, he/she doesn't need to know anything about Android Studio. If that's what you meant.

**Comment by franzsilva** [comment]
Yep exactly, so to make sure: 
if the flag --fuchsia is passed it will only check for 
Flutter version
IntelliJ Plugin

**Comment by xster** [comment]
Sounds good

File: packages/flutter_tools/lib/src/commands/doctor.dart [hunk 1]
```diff
@@ -14,6 +14,11 @@ class DoctorCommand extends FlutterCommand {
       negatable: false,
       help: 'Run the Android SDK manager tool to accept the SDK\'s licenses.',
     );
+    argParser.addFlag('fuchsia',
+      defaultsTo: false,
+      negatable: false,
+      hide: true
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/doctor.dart]: This should be `!verbose` or `!verboseHelp` or something (check the other commands for the pattern), and it should be documented with a `help` argument. We do want it to show up if you do `flutter --help --verbose`

File: packages/flutter_tools/lib/src/doctor.dart [hunk 2]
```diff
@@ -34,14 +35,17 @@ class Doctor {
       _validators = <DoctorValidator>[];
       _validators.add(new _FlutterValidator());
 
-      if (androidWorkflow.appliesToHostPlatform)
-        _validators.add(androidWorkflow);
-
-      if (iosWorkflow.appliesToHostPlatform)
-        _validators.add(iosWorkflow);
+      if (!runOnlyFuchsia){
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/doctor.dart]: nit: missing space between `)` and `{`

File: packages/flutter_tools/lib/src/doctor.dart [hunk 3]
```diff
@@ -95,7 +99,10 @@ class Doctor {
   }
 
   /// Print verbose information about the state of installed tooling.
-  Future<bool> diagnose({ bool androidLicenses: false }) async {
+  Future<bool> diagnose({ bool androidLicenses: false, bool onlyFuchsia: false }) async {
+    runOnlyFuchsia = onlyFuchsia;
+    
+
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/doctor.dart]: extraneous blank line here

**Comment by Hixie** [comment]
test?

**Comment by Hixie** [comment]
@franzsilva Are you still intending to work on this PR?

**Comment by franzsilva** [comment]
Hi @Hixie. Yes. But I need to create a new pull request because I accidentally deleted the fork I was working on. My bad.

**Comment by mit-mit** [comment]
Closing the present PR then. Look forward to a new one!

**Pull request closed without merging**